core/rovecomm.py

```diff
@@ -141,46 +182,157 @@ def read(self):
             return_packet (RoveCommPacket): A RoveCommPacket that contains a
             RoveComm message received over the network
         '''
-        try:
-            packet, remote_ip = self.RoveCommSocket.recvfrom(1024)
-            header_size = struct.calcsize(ROVECOMM_HEADER_FORMAT)
+        ready = select.select([self.RoveCommSocket], [], [], 0)
+        if len(ready[0]) != 0:
```

Comment:
You can just do `if read[0]:`

Reply:
If ready[0] would imply we are looking to see if the element at index 0 evaluates to true, where here we are looking if ready[0] exists/has length. You could just say len(ready[0]), but != 0 is more transparent